collector/generic_collector.go

```diff
@@ -36,6 +36,8 @@ type GenericCollector struct {
 
 	//holds information necessary to extract metrics
 	info *collectorInfo
+
+	metrics []v1.Metric
```

Comment:
There is redundancy here as we store metrics as part of the Collector struct as well as return it in the Collect() method.